### PR TITLE
Amend/update angular material2

### LINKED ISSUE #186: Fix manage ui text overflow
image
Make the dat3es overflow correctly. (Use flexbox and text-overflow: ellipsis)

### LINKED ISSUE #217: TypeError: Die Eigenschaft "initialNavigation" eines undefinierten oder Nullverweises kann nicht abgerufen w...
Happens only in IE on page load

https://sentry.pegnu.cloud/pg/outstanding-cobra/issues/82/

```
TypeError: Die Eigenschaft "initialNavigation" eines undefinierten oder Nullverweises kann nicht abgerufen werden.
  at t.prototype.isLegacyDisabled (~/@angular/router/@angular/router.es5.js:5968)
  at Anonymous function (~/@angular/router/@angular/router.es5.js:5905)
  at t.prototype.invoke (~/zone.js/dist/zone.js:363)
  at onInvoke (~/@angular/core/@angular/core.es5.js:4125)
  at t.prototype.invoke (~/zone.js/dist/zone.js:363)
...
(8 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by M4R1KU**
This closes #217 closes #186.

Summary of changes:
- ​fixed IE bug -> IE works again but still buggy
- updated angular and angular/material2
- refactored some things

I made sure to:
- [x] Add and / or update tests
- [x] Run the code formatter (`CTRL + ALT + L`)
- [x] Update any documentation (javadoc, domain model, erd, ...)
- [x] Follow the contribution guidelines
- [x] Update CI script and configs on Jenkins (`Jenkinsfile`, `auth0.yml`, `application.yml`, etc.)

Please review @jmesserli @needToRoll  / cc: @outcobra/developers.

### CODE REVIEW

File: frontend/src/app/core/common/oc-date-formats.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import {MdDateFormats} from '@angular/material/typings/core/datetime/date-formats';
+// currently useless because there's no date adapter that can handle the parse format
```
Comment (jmesserli) [inline frontend/src/app/core/common/oc-date-formats.ts]: This comment will probably be forgotten and obsolete

File: frontend/src/app/oc-material.module.ts [hunk 2]
```diff
@@ -55,7 +60,15 @@ import {
         MdSliderModule,
         MdRippleModule,
         MdListModule,
+        MdDatepickerModule,
         MdCoreModule
+    ],
+    providers: [
+        //TODO activate this after the next material2 release (https://github.com/angular/material2/pull/4541 : https://github.com/angular/material2/issues/4534)
```
Comment (jmesserli) [inline frontend/src/app/oc-material.module.ts]: Is this not activated yet like this?

Comment (M4R1KU) [inline frontend/src/app/oc-material.module.ts]: Yup that comment is kind of bullshit

File: frontend/src/app/core/services/ui/responsive-helper.service.ts [hunk 3]
```diff
@@ -1,39 +1,37 @@
...
-    public listenForResize(): Observable<any> {
-        return Observable.fromEvent(window, 'resize')
-            .do(() => this._mobile = this._checkMobile());
+    public listenForBreakpointChange(): Observable<OCMediaChange> {
+        return this._mediaChange;
     }
 
-    public listenForOrientationChange(): Observable<Orientation> {
+    public listenForOrientationChange(): Observable<OCMediaChange> {
         return Observable.fromEvent(window, 'orientationchange')
```
Comment (jmesserli) [inline frontend/src/app/core/services/ui/responsive-helper.service.ts]: nice

File: frontend/src/app/core/util/helper.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ export function isFalsy(toCheck: any): boolean {
 }
 
 export function isNull(toCheck: any): boolean {
```
Comment (jmesserli) [inline frontend/src/app/core/util/helper.ts]: `isNullOrUndefined`

Comment (M4R1KU) [inline frontend/src/app/core/util/helper.ts]: To long :roll_eyes:

File: frontend/src/app/core/services/ui/responsive-helper.service.ts (hunk 3 above)
Comment (M4R1KU) [inline frontend/src/app/core/services/ui/responsive-helper.service.ts]: ?

File: frontend/src/app/core/util/helper.ts (hunk 4 above)
Comment (jmesserli) [inline frontend/src/app/core/util/helper.ts]: Long or wrong? ;)

File: frontend/src/app/core/services/ui/responsive-helper.service.ts (hunk 3 above)
Comment (jmesserli) [inline frontend/src/app/core/services/ui/responsive-helper.service.ts]: Not relevant but it's nice

File: frontend/src/app/core/util/helper.ts (hunk 4 above)
Comment (M4R1KU) [inline frontend/src/app/core/util/helper.ts]: Long

File: frontend/src/app/core/services/ui/responsive-helper.service.ts (hunk 3 above)
Comment (M4R1KU) [inline frontend/src/app/core/services/ui/responsive-helper.service.ts]: Don't get it

Comment (jmesserli) [inline frontend/src/app/core/services/ui/responsive-helper.service.ts]: Just forget it

**Comment by needToRoll** [comment]
I just deployed this branch to firebase (to take a look at the dialog issues, and a possible fix) and it seems like we have an issue with mobile compatibility.
On my Iphone the screen stays empty if i try to open the page.
Please take a look before merging this.

**Comment by M4R1KU** [comment]
@needToRoll did you cleared the cache? It solved some issues for me though. Dialogs do not seem to be broken (at least in the emulator) and I am currently to tired to get up and take my iPhone xD.

**Comment by needToRoll** [comment]
Well according to #195 they need to be realigned... the other parts of the issue are resolved imo

**Comment by M4R1KU** [comment]
I am aware of #195 but I do not have a good idea currently :woman_shrugging:

**Comment by needToRoll** [comment]
The vertical alignment can be fixed very easy. Just change the constants for the MOBILE_DIALOG.
And for the buttons, I think yes or no is fine... we don't need the third option.

**Comment by M4R1KU** [comment]
Ah I forgot the buttons

**Comment by M4R1KU** [comment]
do believe that it is good or do you want to re-review? :p

**Comment by needToRoll** [comment]
ng build is still failing

**Comment by M4R1KU** [comment]
Tests are green so merge ready? @needToRoll @jmesserli

**Pull request merged**